Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/BookingExtra.hs

```diff
@@ -71,6 +71,10 @@ findByTransactionId txnId =
     (Just (Se.Desc BeamB.createdAt))
     <&> listToMaybe
 
+findByTransactionIdAndStatus :: (MonadFlow m, EsqDBFlow m r, CacheFlow m r) => Text -> BookingStatus -> m (Maybe Booking)
```

Comment:
Will have to confirm if this query performs well, 
or we should only lookup for given `transactionId` and then filter based on status.